### ISSUE TITLE
Comparing ETS with Statsmodels ExponentialSmoothing

### ISSUE THREAD

**Issue by hrzn** [Nixtla/statsforecast#211]
Hi guys,

I was playing a little with `ETS` to see whether we could include it in Darts.
For the time being I'm having a hard time to have it outperform `statsmodels` in terms of runtime (I haven't looked at accuracy). Is there any special trick (or a special regime to be in) in order for the `statsforcast` version to run faster?

Here is the small benchmark I ran:
```python
import numpy as np
import time

from statsforecast.models import ETS as SF_ETS
import statsmodels.tsa.holtwinters as hw

values = np.random.rand(100,)

# First, we make a dry run for jit:
model_sf = SF_ETS()
model_sf.fit(values)
_ = model_sf.predict(10)

model_sm = hw.ExponentialSmoothing(values)
sm_res = model_sm.fit()
_ = sm_res.forecast(10)

# Now, a small benchmark:
tic = time.time()
for _ in range(100):
    model_sf = SF_ETS()
    model_sf.fit(values)
    _ = model_sf.predict(10)
    
print("Time taken by SF ETS : {:.2f} s.".format(time.time() - tic))

tic = time.time()
for _ in range(100):
    model_sm = hw.ExponentialSmoothing(values)
    sm_res = model_sm.fit()
    _ = sm_res.forecast(10)
    
print("Time taken by SM ETS : {:.2f} s.".format(time.time() - tic))
```

And I get results like
```
Time taken by SF ETS : 1.54 s.
Time taken by SM ETS : 0.38 s.
```

**Comment by kdgutier** [comment]
Hi @hrzn,

The speed of the method was an issue for other users too, maybe this solution can help: https://github.com/Nixtla/statsforecast/issues/84.

I think that that time precompilation of `numba` code made the difference.

**Comment by hrzn** [comment]
> Hi @hrzn,
> 
> The speed of the method was an issue for other users too, maybe this solution can help: #84.
> 
> I think that that time precompilation of `numba` code made the difference.

Thanks for your quick reply @kdgutier. I've taken care of this jit compilation aspect in my benchmark (see the dry-run pass before the actual benchmark in my code). So I don't think that's the issue.

**Comment by kdgutier** [comment]
Hi @hrzn,

I think that you are comparing the speed of a single exponential smoothing versus the speed of an optimized exponential smoothing (the search normally takes 94 models).

**Comment by hrzn** [comment]
Oh, so if I understand well you're saying that `statsforecast.models.ETS` is in fact doing "Auto-ETS"? Over what hyper-parameters?

**Comment by kdgutier** [comment]
> Oh, so if I understand well you're saying that `statsforecast.models.ETS` is in fact doing "Auto-ETS"? Over what hyper-parameters?

Hi @hrzn yes, I give the word to @FedericoGarza. He is writing an explanation.

**Comment by hrzn** [comment]
Awesome, thanks!

**Comment by FedericoGarza** [comment]
Hi @hrzn! Yes, `statsforecast.models.ETS` is performing "`AutoETS`". 

From what I could see in the documentation, `statsmodels.tsa.holtwinters.ExponentialSmoothing` fits a simple exponential smoothing (no trend and no seasonality). You can fit this model using `model='ANN'` (additive error, no trend, and no seasonality), as follows,

```python
import numpy as np
import time

from statsforecast.models import ETS as SF_ETS
import statsmodels.tsa.holtwinters as hw

values = np.random.rand(100,)

# First, we make a dry run for jit:
model_sf = SF_ETS(model='ANN')
model_sf.fit(values)
_ = model_sf.predict(10)

model_sm = hw.ExponentialSmoothing(values)
sm_res = model_sm.fit()
_ = sm_res.forecast(10)

# Now, a small benchmark:
tic = time.time()
for _ in range(100):
    model_sf = SF_ETS(model='ANN')
    model_sf.fit(values)
    _ = model_sf.predict(10)
    
print("Time taken by SF ETS : {:.2f} s.".format(time.time() - tic))

tic = time.time()
for _ in range(100):
    model_sm = hw.ExponentialSmoothing(values)
    sm_res = model_sm.fit()
    _ = sm_res.forecast(10)
    
print("Time taken by SM ETS : {:.2f} s.".format(time.time() - tic))
```

And the results are,

```
Time taken by SF ETS : 0.11 s.
Time taken by SM ETS : 0.31 s.
```

**Comment by hrzn** [comment]
Very nice, thanks @FedericoGarza. Now, if I specify `model` (e.g., `model="ANN"`), is there still some hyper-parameter optimization going on?

**Comment by kdgutier** [comment]
Search is performed against AICc and single models optimize the normal likelihood.

**Comment by FedericoGarza** [comment]
Exactly @hrzn, there is no further hyperparameter optimization once you define the model's components. In addition, you can define the component on which you want to perform the hyperparameter optimization. For example, `model='AAZ'` will use an additive error and additive trend and choose the kind of seasonality.

**Comment by hrzn** [comment]
Thanks for the explanations! So I assume the "Z" character is a placeholder to ask the model to figure out the best parameter, is that correct?
In any case this is great - we'll include it in Darts :)

**Comment by FedericoGarza** [comment]
Yes, @hrzn, that's correct :)

Nice! Let us know if we can be of any help.

**Comment by hrzn** [comment]
I will thanks :) closing this issue as I think you guys answered all my questions.
